[Workers] wrangler get started update (#6874)
cloudflare/cloudflare-docs: content/workers/wrangler/install-and-update.md

---
pcx_content_type: how-to
title: Install/Update Wrangler
weight: 1
---

# Install/Update Wrangler

{{<Aside type="note">}}

> Wrangler has launched a new version. If you previously had Wrangler 1 installed or were working on a Wrangler 1 project, refer to the [Migration guide](/workers/wrangler/migration/migrating-from-wrangler-1/).

Reviewer: ```suggestion
There is a new version of Wrangler. If you previously had Wrangler 1 installed or were working on a Wrangler 1 project, refer to the [Migration guide](/workers/wrangler/migration/migrating-from-wrangler-1/).
```

{{</Aside>}}

Wrangler is a command-line tool for building [Cloudflare Workers](https://workers.cloudflare.com/).

## Install Wrangler

To install [Wrangler](https://github.com/cloudflare/wrangler2), ensure you have [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/getting-started) installed, preferably using a Node version manager like [Volta](https://volta.sh/) or [nvm](https://github.com/nvm-sh/nvm). Using a version manager helps avoid permission issues and allows you to easily change Node.js versions. Wrangler requires a Node version of `16.13.0` or later. After you have installed npm and Node.js, run:

```sh
$ npm install -g wrangler
```

or install with `yarn`:

```sh
$ yarn global add wrangler
```

## Update Wrangler

> To check your Wrangler version, run:

Reviewer: I would make this something like:

If there is a package.json in the current directory, and wrangler is in it:
- `npm upgrade wrangler --save` or `npm update wrangler --save`

If outside of a project folder (and there is no parent directory containing a package.json with wrangler in it:
- `npm update -g wrangler`

Caveat: `npx wrangler@latest` will use the latest version of Wrangler *except* in the case where there's already a Wrangler version installed in the current folder, then it'll use that version instead.

---

The caveat is important - Wrangler deviates from standard npm package behaviour by doing this

---
pcx_content_type: how-to
title: Install/Update Wrangler
weight: 1
---

# Install/Update Wrangler

{{<Aside type="note">}}

This is an installation guide for the new version of Wrangler. If you previously had Wrangler 1 installed or were working on a Wrangler 1 project, refer to the [Migration guide](/workers/wrangler/migration/migrating-from-wrangler-1/).

{{</Aside>}}

Wrangler is a command-line tool for building with Cloudflare developer products.

## Install Wrangler

> To install [Wrangler](https://github.com/cloudflare/wrangler2), ensure you have [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/getting-started) installed, preferably using a Node version manager like [Volta](https://volta.sh/) or [nvm](https://github.com/nvm-sh/nvm). Using a version manager helps avoid permission issues and allows you to easily change Node.js versions. Wrangler requires a Node version of `16.13.0` or later. 

Reviewer 2: Maybe we should use "Node" or "Node.js", but not both (the reader may not know if they're the same thing). Alternatively, say that they're the same thing.

> To install [Wrangler](https://github.com/cloudflare/wrangler2), ensure you have [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/getting-started) installed, preferably using a Node version manager like [Volta](https://volta.sh/) or [nvm](https://github.com/nvm-sh/nvm). Using a version manager helps avoid permission issues and allows you to easily change Node.js versions. Wrangler requires a Node version of `16.13.0` or later. 

Author: ```suggestion
To install [Wrangler](https://github.com/cloudflare/wrangler2), ensure you have [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/getting-started) installed, preferably using a Node version manager like [Volta](https://volta.sh/) or [nvm](https://github.com/nvm-sh/nvm). Using a version manager helps avoid permission issues and allows you to easily change Node.js versions. Wrangler requires a Node.js version of `16.13.0` or later. 
```

### Install Wrangler globally

> After you have installed npm and Node.js, run:

Reviewer 2: Should we use monospace formatting for `npm`, since we're doing it for `yarn` (a few lines below)?

```sh
$ npm install -g wrangler
```

or install with `yarn`:

```sh
$ yarn global add wrangler
```

### Install Wrangler locally

> To install a local version of Wrangler, run:

Reviewer: ```suggestion
To install Wrangler locally within your Worker project, run:
```

```sh
$ npm install wrangler
```

or install with `yarn`:

```sh
$ yarn add wrangler
```

## Update Wrangler

To check your Wrangler version, run:

```sh
$ wrangler version
```

### Update Wrangler globally

To update Wrangler globally, you must be outside of a project folder (and there is no parent directory containing a `package.json`) and run:

```sh
$ npm update -g wrangler
```

### Update Wrangler locally

To update Wrangler only in your current directory containing a `package.json`, run:

```sh
$ npm upgrade wrangler --save
```

or

```sh
$ npm update wranger --save
```

{{<Aside type="warning" header="`npx wrangler innit`">}}

> `npx wrangler` will use the latest version of Wrangler except when there is already a Wrangler version installed in the current folder. In this case, it will use that version instead. Running `npx wrangler init` will install a local version of Wrangler in the project directory.

Reviewer: ```suggestion
`npx wrangler` will use the latest version of Wrangler except when there is already a Wrangler version installed in the current folder. In this case, it will use the locally installed version instead. Running `npx wrangler init` will install a local version of Wrangler in the newly created project directory.
```

> `npx wrangler` will use the latest version of Wrangler except when there is already a Wrangler version installed in the current folder. In this case, it will use the locally installed version instead. Running `npx wrangler init` will install a local version of Wrangler in the newly created project directory.

Reviewer 2: ```suggestion
Running `npx wrangler` will use the latest version of Wrangler except when there is already a Wrangler version installed in the current folder. In this case, it will use the locally installed version instead. Running `npx wrangler init` will install a local version of Wrangler in the newly created project directory.
```
Trying to avoid a monospaced piece of text at the beginning of the sentence.
